components/camel-elytron/src/main/java/org/apache/camel/component/elytron/ElytronComponent.java

```diff
@@ -0,0 +1,163 @@
+import org.wildfly.security.http.HttpConstants;
+import org.wildfly.security.http.HttpServerAuthenticationMechanismFactory;
+import org.wildfly.security.http.util.FilterServerMechanismFactory;
+import org.wildfly.security.http.util.SecurityProviderServerMechanismFactory;
+
+import javax.net.ssl.SSLContext;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.Provider;
+import java.util.Collections;
+
+/**
+ * Elytron component brings elytron security over came-undertow component.
+ *
+ * Component work either as producer and as consumer.
+ *
+ * User has to define its SecurityDomain.Builder which will be used for creation of security domain.
+ * MechanismName then allows to define mechanism, which will take care of authentication from security context.
+ * MechanusmName should be selected with regard to default securityRealm.
+ *
+ * Example: to use bearer_token, mechanism name has to be "BEARER_TOKEN" and realm has to be TokenSecurityRealm.
+ *
+ * @author JiriOndrusek
+ */
+@Metadata(label = "verifiers", enums = "parameters,connectivity")
+@Component("elytron")
+public class ElytronComponent extends UndertowComponent {
+
+    private static final WildFlyElytronProvider elytronProvider = new WildFlyElytronProvider();
+
+    @Metadata(label = "elytron")
+    private SecurityDomain.Builder securityDomainBuilder;
+    @Metadata(label = "elytron", defaultValue = HttpConstants.BEARER_TOKEN)
+    private String mechanismName;
+
+    private SecurityDomain securityDomain;
+
+    public ElytronComponent() {
+    }
+
+    public ElytronComponent(CamelContext context) {
+        super(context);
+    }
+
+    @Override
+    protected String getComponentName() {
+        return "elytron";
+    }
+
+
+    @Override
+    protected UndertowEndpoint createEndpointInstance(URI endpointUri, UndertowComponent component) throws URISyntaxException {
+        return new ElytronEndpoint(endpointUri.toString(), component);
+    }
+
+    @Override
+    public HttpHandler registerEndpoint(HttpHandlerRegistrationInfo registrationInfo, SSLContext sslContext, HttpHandler handler) {
+        //injecting elytron
+        return super.registerEndpoint(registrationInfo, sslContext, wrap(handler, getSecurityDomain()));
+    }
+
+    /**
+     * Definition of Builder, which will be used for creation of security domain.
+     */
+    public SecurityDomain.Builder getSecurityDomainBuilder() {
+        return securityDomainBuilder;
+    }
+
+    public void setSecurityDomainBuilder(SecurityDomain.Builder securityDomainBuilder) {
+        this.securityDomainBuilder = securityDomainBuilder;
+    }
+
+
+    /**
+     * Name of the mechanism, which will be used for selection of authentication mechanism.
+     */
+    public String getMechanismName() {
+        return mechanismName;
+    }
+
+    public void setMechanismName(String mechanismName) {
+        this.mechanismName = mechanismName;
+    }
+
+    SecurityDomain getSecurityDomain() {
```

Comment:
Can we add the access modifier here?

Reply:
Would "protected" suffice? I'm accessing this getter from endpoint

Reply:
sorry didn't notice, should be fine I guess.